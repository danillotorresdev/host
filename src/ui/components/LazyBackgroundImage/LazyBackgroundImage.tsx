import { useState } from "react";

type LazyBackgroundImageProps = {
  img: string;
  children?: JSX.Element[] | JSX.Element;
  style?: React.CSSProperties;
  isDarkened?: boolean;
  className?: string;
};

export function LazyBackgroundImg({
  img,
  children,
  style,
  isDarkened,
  className,
}: Readonly<LazyBackgroundImageProps>) {
  const [loaded, setLoaded] = useState(false);

  const handleLoad = () => {
    setLoaded(true);
  };

  return (
    <div
      className={className}
      style={{
        backgroundImage: `${
          isDarkened
            ? "linear-gradient( rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6) ),"
            : ""
        }url(${img})`,
        backgroundRepeat: "no-repeat",
        backgroundPosition: "center center",
        backgroundSize: "cover",
        filter: loaded ? "none" : "blur(20px)",
        transition: "filter 0.5s",
        width: "100%",
        height: "100%",
        ...style,
      }}
    >
      <img src={img} alt="" onLoad={handleLoad} style={{ display: "none" }} />
      {loaded && children}
    </div>
  );
}