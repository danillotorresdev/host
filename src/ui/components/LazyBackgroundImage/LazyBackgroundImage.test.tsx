
import { LazyBackgroundImg } from './LazyBackgroundImage';
import { render } from "@/utils/settings/tests/utilities";
import { expect, it, describe } from "vitest";

describe("LazyBackgroundImg", () => {
  it("should render the user's name", () => {
    const { asFragment } = render(
      <LazyBackgroundImg
        img="https://avatars.githubusercontent.com/u/1?v=4"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  }
  )
});
