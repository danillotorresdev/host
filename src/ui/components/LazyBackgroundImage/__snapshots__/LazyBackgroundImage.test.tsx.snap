// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LazyBackgroundImg > should render the user's name 1`] = `
<DocumentFragment>
  <div
    style="background-image: url(https://avatars.githubusercontent.com/u/1?v=4); background-repeat: no-repeat; background-position: center center; background-size: cover; filter: blur(20px); transition: filter 0.5s; width: 100%; height: 100%;"
  >
    <img
      alt=""
      src="https://avatars.githubusercontent.com/u/1?v=4"
      style="display: none;"
    />
  </div>
</DocumentFragment>
`;
